### DISCUSSION TITLE
icubsrv & icub-head password and the correct env parameter YARP_ROBOT_NAME

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#703] [Q&A] [answered]
Hi,

Upon running the icubsrv & icub-head nodes following the launch of `yarpmanager`, I encounter a dialog box requesting me to enter a password as belowed, which was not required previously.

Does anyone know what the issue might be and how to eliminate the need for entering the password?

Additionally, I have another question: What is the correct environmental parameter `YARP_ROBOT_NAME` for an Ubuntu system on a laptop? Currently my system is `icubGazeboSim`

Thanks.
```
icub@icubsrv:~$ yarpmanager
[ERROR] |yarp.os.Property| cannot read from ymanager.ini
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://10.0.0.1:10002/
[WARNING] Cannot connect to /icubsrv
[ERROR] ClusterWidget: port /icubsrv is not responding
[INFO] |yarp.os.Port|/tmp/port/1| Port /tmp/port/1 active at tcp://10.0.0.1:10002/
[WARNING] Cannot connect to /icub-head
[ERROR] ClusterWidget: port /icub-head is not responding
icub@icubsrv's password: 
icub@icub-head's password:
```

**Comment by Nicogene** [comment, marked as answer]
Hi @jimingda, `ClusterWidget` spawn the `yarprun`s via ssh, if you want to be able to not insert every time the password you have to exchange the public keys between hosts:
- https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/icub-server-laptop/?h=ssh#install-the-ssh-keys-for-password-less-login-on-icub-head

> Additionally, I have another question: What is the correct environmental parameter YARP_ROBOT_NAME for an Ubuntu system on a laptop? Currently my system is icubGazeboSim

Usually, that variable is used on `icub-head`, if the laptop is the one used with the robot I advise to use the same `YARP_ROBOT_NAME`, otherwise it can be left empty
